src/js/plugins/model.ts

```diff
@@ -0,0 +1,53 @@
+import Plugin from 'plugins/plugin';
+import { log } from 'utils/log';
+import { getPluginName } from 'plugins/utils';
+import type { PluginObj } from 'types/generic.type';
+
+export interface PluginModelInt extends Function {
+    addPlugin: (url: string) => PluginObj;
+    setupPlugin: (url: string) => Promise<PluginObj>;
+    getPlugin: (name: string) => PluginObj;
+    removePlugin: (name: string) => void;
+    getPlugins: () => { [key: string]: PluginObj };
+}
+
+const pluginsRegistered: { [key: string]: PluginObj } = {};
+
+const PluginModel = function(this: PluginModelInt): void {
```

Comment:
i moved this into a function vs using a prototype so that i could make all the functions on it required, otherwise since the prototype added them after the function creation i would had to have made them all optional, no bueno

Reply:
Making it a class would be a cleaner refactor than this

Reply:
class-ified and updated